zerver/lib/markdown/__init__.py

```diff
@@ -1342,54 +1344,27 @@ def handleMatch(self, match: Match[str]) -> Optional[Element]:
         return time_element
 
 
-# All of our emojis (excluding ZWJ sequences) belong to one of these Unicode blocks:
-# \U0001f100-\U0001f1ff - Enclosed Alphanumeric Supplement
-# \U0001f200-\U0001f2ff - Enclosed Ideographic Supplement
-# \U0001f300-\U0001f5ff - Miscellaneous Symbols and Pictographs
-# \U0001f600-\U0001f64f - Emoticons (Emoji)
-# \U0001f680-\U0001f6ff - Transport and Map Symbols
-# \U0001f7e0-\U0001f7eb - Coloured Geometric Shapes (NOTE: Not Unicode standard category name)
-# \U0001f900-\U0001f9ff - Supplemental Symbols and Pictographs
-# \u2000-\u206f         - General Punctuation
-# \u2300-\u23ff         - Miscellaneous Technical
-# \u2400-\u243f         - Control Pictures
-# \u2440-\u245f         - Optical Character Recognition
-# \u2460-\u24ff         - Enclosed Alphanumerics
-# \u2500-\u257f         - Box Drawing
-# \u2580-\u259f         - Block Elements
-# \u25a0-\u25ff         - Geometric Shapes
-# \u2600-\u26ff         - Miscellaneous Symbols
-# \u2700-\u27bf         - Dingbats
-# \u2900-\u297f         - Supplemental Arrows-B
-# \u2b00-\u2bff         - Miscellaneous Symbols and Arrows
-# \u3000-\u303f         - CJK Symbols and Punctuation
-# \u3200-\u32ff         - Enclosed CJK Letters and Months
-UNICODE_EMOJI_RE = (
-    "(?P<syntax>["
-    "\U0001F100-\U0001F64F"
-    "\U0001F680-\U0001F6FF"
-    "\U0001F7E0-\U0001F7EB"
-    "\U0001F900-\U0001F9FF"
-    "\u2000-\u206F"
-    "\u2300-\u27BF"
-    "\u2900-\u297F"
-    "\u2B00-\u2BFF"
-    "\u3000-\u303F"
-    "\u3200-\u32FF"
-    "])"
+# From https://unicode.org/reports/tr51/#EBNF_and_Regex. Keep this synced with `possible_emoji_regex`.
+POSSIBLE_EMOJI_RE = regex.compile(
+    r"""^(.*?)(?P<syntax>
+\p{RI} \p{RI}
+| \p{Emoji}
+  ( \p{EMod}
```

Comment:
All these `(`s need to be converted to non-capturing `(?:`.

Reply:
Done :)

Reply:
Added! Curious why it doesn't work to add to the `(.*?)` blocks - are those parentheses semantically different?

Reply:
The two capturing `(.*?)` blocks are required for the way Python-Markdown uses this regex ([documentation](https://python-markdown.github.io/extensions/api/#inline-patterns)). They are normally added by the `markdown.inlinepatterns.Pattern` constructor that our `CompiledPattern` constructor is skipping, so we need to provide them to provide them to `CompiledPattern` ourselves. Non-named capturing blocks are numbered, so inserting any other non-named capturing blocks between them breaks the numbering that Python-Markdown expects.